test/system_test.py

```diff
@@ -145,37 +145,40 @@ def test_install_version_1_1_3(self):
         spack_install_and_test('gridtools @1.1.3')
 
 
-@pytest.mark.no_tsa  # config file does not exist for these machines
+@pytest.mark.no_tsa  # config file does not exist for this machines
 class IconTest(unittest.TestCase):
 
     def test_install_nwp_gpu(self):
         spack_install_and_test(
-            'icon @nwp %nvhpc icon_target=gpu +claw +eccodes +ocean')
+            f'icon @nwp %nvhpc icon_target=gpu ^{mpi} %{nvidia_compiler}')
 
     def test_install_nwp_cpu(self):
         spack_install_and_test(
-            'icon @nwp %nvhpc icon_target=cpu serialize_mode=create +eccodes +ocean'
-        )
+            f'icon @nwp %nvhpc icon_target=cpu ^{mpi} %{nvidia_compiler}')
 
-    def test_devbuild_nwp_gpu(self):
-        spack_install_and_test(
-            'icon @develop %nvhpc config_dir=./.. icon_target=gpu')
+    # def test_devbuild_nwp_gpu(self):
+    #     spack_install_and_test(
+    #         f'icon @develop %nvhpc config_dir=./.. icon_target=gpu ^{mpi} %{nvidia_compiler}')
 
-    def test_devbuild_nwp_cpu(self):
-        spack_install_and_test(
-            'icon @develop %nvhpc config_dir=./.. icon_target=cpu')
+    # def test_devbuild_nwp_cpu(self):
+    #     spack_install_and_test(
+    #         f'icon @develop %nvhpc config_dir=./.. icon_target=cpu ^{mpi} %{nvidia_compiler}')
 
+    @pytest.mark.no_balfrin  # config file does not exist for this machines
     def test_install_exclaim_cpu(self):
         spack_install_and_test(
-            'icon @exclaim-master %nvhpc icon_target=cpu +eccodes +ocean')
+            f'icon @exclaim-master %nvhpc icon_target=cpu +eccodes +ocean ^{mpi} %{nvidia_compiler}'
+        )
 
+    @pytest.mark.no_balfrin  # config file does not exist for this machines
     def test_install_exclaim_cpu_gcc(self):
         spack_install_and_test(
             'icon @exclaim-master %gcc icon_target=gpu +eccodes +ocean +claw')
```

Comment:
```suggestion
            'icon @exclaim-master %gcc icon_target=cpu +eccodes +ocean +claw')
```

Reply:
I will fix this in #602 to avoid conflicts.